english/tests/tests.py

```diff
@@ -5,121 +5,292 @@
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class LessonsForStudent(APITestCase):
     fixtures = [
         "english/tests/fixtures.json",
     ]
 
     def setUp(self):
-        self.testUser = User.objects.get(id=1)
-        self.client.force_login(user=self.testUser)
+        self.student = User.objects.get(id=1)
+        self.teacher = User.objects.get(id=2)
 
-    def test_get_lessons_for_current_student(self):
+    def test_get_lessons_success(self):
+        self.client.force_login(user=self.student)
         expected_data = {
             "pk": 1,
-            "class_name": 1,
-            "_status": "PROGRESS",
-            "time_start": "2022-07-04T01:14:43",
-            "time_end": "2022-07-30T01:14:46"
+            "class_name": "FirstClass",
+            "_status": "COMING",
+            "time_start": "2030-07-04T01:14:43",
+            "time_end": "2030-10-30T01:14:46"
         }
-        url = reverse("student-lessons")
+        url = reverse("student-lessons-list")
         response = self.client.get(url)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertEqual(len(response.data), 1)
+        self.assertEqual(len(response.data), 2)
         self.assertDictEqual(dict(response.data[0]), expected_data)
 
-    def test_get_lessons_for_current_teacher(self):
+    def test_get_lessons_failed(self):
+        self.client.force_login(user=self.teacher)
+        url = reverse("student-lessons-list")
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class LessonsForTeacher(APITestCase):
+    fixtures = [
+        "english/tests/fixtures.json",
+    ]
+
+    def setUp(self):
+        self.student = User.objects.get(id=1)
+        self.teacher = User.objects.get(id=2)
+
+    def test_get_lessons_success(self):
+        self.client.force_login(user=self.teacher)
         expected_data = {
-            "pk": 2,
-            "class_name": 2,
-            "_status": "PROGRESS",
-            "time_start": "2022-07-04T01:14:43",
-            "time_end": "2022-07-30T01:14:46"
+            "pk": 1,
+            "class_name": "FirstClass",
+            "_status": "COMING",
+            "time_start": "2030-07-04T01:14:43",
+            "time_end": "2030-10-30T01:14:46"
         }
-        url = reverse("teacher-lessons")
+        url = reverse("teacher-lessons-list-create")
         response = self.client.get(url)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(len(response.data), 1)
         self.assertDictEqual(dict(response.data[0]), expected_data)
+
+    def test_get_lessons_failed(self):
+        self.client.force_login(user=self.student)
+        url = reverse("teacher-lessons-list-create")
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
```

Comment:
Need to add tests for creating lessons: success and failed(403 and incorrect start and end time)